x-pack/plugins/ml/public/application/timeseriesexplorer/components/entity_control/entity_control.tsx

```diff
@@ -129,13 +224,146 @@ export class EntityControl extends Component<EntityControlProps, EntityControlSt
         onSearchChange={this.onSearchChange}
         isClearable={false}
         renderOption={this.renderOption}
         data-test-subj={`mlSingleMetricViewerEntitySelection ${entity.fieldName}`}
+        prepend={
+          <EuiPopover
+            ownFocus
+            button={
+              <EuiButtonIcon
+                color="text"
+                iconSize="xxl"
+                iconType="gear"
+                aria-label={i18n.translate('xpack.ml.timeSeriesExplorer.editControlConfiguration', {
+                  defaultMessage: 'Edit field configuration',
+                })}
+                onClick={() => {
+                  this.setState({
+                    isEntityConfigPopoverOpen: !this.state.isEntityConfigPopoverOpen,
+                  });
+                }}
+                data-test-subj={`mlSingleMetricViewerEntitySelectionConfigButton_${entity.fieldName}`}
+              />
+            }
+            isOpen={this.state.isEntityConfigPopoverOpen}
+            closePopover={() => {
+              this.setState({
+                isEntityConfigPopoverOpen: false,
+              });
+            }}
+          >
+            <div
+              data-test-subj={`mlSingleMetricViewerEntitySelectionConfigPopover_${entity.fieldName}`}
+            >
+              <EuiFlexGroup gutterSize={'xs'} alignItems={'center'}>
+                <EuiFlexItem grow={false}>
+                  <EuiSwitch
+                    label={
+                      <FormattedMessage
+                        id="xpack.ml.timeSeriesExplorer.anomalousOnlyLabel"
+                        defaultMessage="Anomalous only"
+                      />
+                    }
+                    checked={!!this.props.config?.anomalousOnly}
+                    onChange={(e) => {
+                      const isAnomalousOnly = e.target.checked;
+                      this.props.onConfigChange(this.props.entity.fieldType, {
+                        anomalousOnly: isAnomalousOnly,
+                        sort: {
+                          order: this.props.config.sort.order,
+                          by: this.props.config.sort.by,
+                        },
+                      });
+                    }}
+                    compressed
+                    data-test-subj={`mlSingleMetricViewerEntitySelectionConfigAnomalousOnly_${entity.fieldName}`}
+                  />
+                </EuiFlexItem>
+
+                <EuiFlexItem grow={false} style={{ width: '16px', height: '24px' }}>
+                  {!this.props.config?.anomalousOnly ? (
+                    <EuiToolTip
+                      position="top"
+                      content={
+                        this.props.isModelPlotEnabled ? (
+                          <FormattedMessage
+                            id="xpack.ml.timeSeriesExplorer.nonAnomalousResultsWithModelPlotInfo"
+                            defaultMessage="The list contains values for the model plot results."
```

Comment:
I suggest changing to this ( or "obtained from the..."):
```suggestion
                            defaultMessage="The list contains values from the model plot results."
```